src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -551,15 +543,17 @@ CHIP_ERROR TimeSynchronizationServer::SetTimeZone(const DataModel::DecodableList
 
     if (lastTzState != TimeState::kInvalid)
     {
-        const auto & tzStore = GetTimeZone()[0];
-        lastTz.offset        = tzStore.timeZone.offset;
-        if (tzStore.timeZone.name.HasValue())
+        const TimeSyncDataProvider::TimeZoneStore & tzStore = GetTimeZone()[0];
+        lastTz.offset                                       = tzStore.timeZone.offset;
+        if (tzStore.timeZone.name.HasValue() && sizeof(name) >= sizeof(tzStore.name))
```

Comment:
ok, so there's a resize(?) line in GetTimeZone()
    mTimeZoneObj.timeZoneList = mTimeZoneObj.timeZoneList.SubSpan(0, mTimeZoneObj.validSize);

Why not just move that re-size into where the list(s) are actually re-sized? Then the list is properly resized after the call to UpdateTimeZoneState and the getter function has no side effects. At that point, you're just directly using a class-scoped object with (I think) a class-owned buffer that won't go out of scope. And you can just directly use it as lastTz without making a copy.

Reasoning here - 1) Getters with side effects are misleading and 2) memcpy is a big bug source and makes me nervous.

If that's too much for this change, then can this at least be swapped over to use CopyCharSpanToMutableCharSpan?

Reply:
why would you need to re-size in multiple places if you move the resize into the update?